lib/Team.js

```diff
@@ -0,0 +1,172 @@
+    * @param {Requestable.callback} [cb] - will receive the list of repositories
+    * @return {Promise} - the promise for the http request
+    */
+   getRepos(cb) {
+      log(`Fetching repositories for Team ${this.__teamId}`);
+      return this._requestAllPages(`/teams/${this.__teamId}/repos`, undefined, cb);
+   }
+
+   /**
+    * Edit Team information
+    * @see https://developer.github.com/v3/orgs/teams/#edit-team
+    * @param {object} options - Parameters for team edit
+    * @param {string} options.name - The name of the team
+    * @param {string} [options.description] - Team description
+    * @param {string} [options.repo_names] - Repos to add the team to
+    * @param {string} [options.privacy=secret] - The level of privacy the team should have. Can be either one
+    * of: `secret`, or `closed`
+    * @param {Requestable.callback} [cb] - will receive the updated team
+    * @return {Promise} - the promise for the http request
+    */
+   editTeam(options, cb) {
+      log(`Editing Team ${this.__teamId}`);
+      return this._request('PATCH', `/teams/${this.__teamId}`, options, cb);
+   }
+
+   /**
+    * List the users who are members of the Team
+    * @see https://developer.github.com/v3/orgs/teams/#list-team-members
+    * @param {object} options - Parameters for listing team users
+    * @param {string} [options.role=all] - can be one of: `all`, `maintainer`, or `member`
+    * @param {Requestable.callback} [cb] - will receive the list of users
+    * @return {Promise} - the promise for the http request
+    */
+   listMembers(options, cb) {
+      log(`Getting members of Team ${this.__teamId}`);
+      return this._requestAllPages(`/teams/${this.__teamId}/members`, options, cb);
+   }
+
+   /**
+    * Get Team membership status for a user
+    * @see https://developer.github.com/v3/orgs/teams/#get-team-membership
+    * @param {string} username - can be one of: `all`, `maintainer`, or `member`
+    * @param {Requestable.callback} [cb] - will receive the membership status of a user
+    * @return {Promise} - the promise for the http request
+    */
+   getMembership(username, cb) {
+      log(`Getting membership of user ${username} in Team ${this.__teamId}`);
+      return this._request('GET', `/teams/${this.__teamId}/memberships/${username}`, undefined, cb);
+   }
+
+   /**
+    * Add a member to the Team
+    * @see https://developer.github.com/v3/orgs/teams/#add-team-membership
+    * @param {string} username - can be one of: `all`, `maintainer`, or `member`
+    * @param {object} options - Parameters for adding a team member
+    * @param {string} [options.role=member] - The role that this user should have in the team. Can be one
+    * of: `member`, or `maintainer`
+    * @param {Requestable.callback} [cb] - will receive the membership status of added user
+    * @return {Promise} - the promise for the http request
+    */
+   addUser(username, options, cb) {
```

Comment:
I just realized that I missed a test for the membership removal, working on that now.